src/box/lua/info.c

```diff
@@ -650,6 +663,22 @@ lbox_schema_version(struct lua_State *L)
 	return 1;
 }
 
+/* gethostname() Lua interface inside box.info. */
+static int
+lbox_info_hostname(struct lua_State *L)
+{
+	char buffer[TT_HOST_NAME_MAX + 1];
```

Comment:
There is no yield between the `gethostname()` and `lua_pushstring()` and all calls are from the TX threads. We can use `static` for the buffer.

Reply:
There was similar issue when somebody tried to use static buffers in GC callback.

E.g. https://github.com/tarantool/tarantool/commit/c302360f0e27b6c1c30eb5787557bc5949db1336#diff-01cf72a8b25f913884f0b9d67ec342db4828a0d27490241f36e969db766f2f48R96

Reply:
Ouch, yes. If a GC callback invokes `box.info.hostname`, we rewrite the buffer before `lua_pushstring()` catches it.

It would be a strange GC callback, but possible. Okay, let's allocate it on the stack on each call.